### DISCUSSION TITLE
Kernel selection

### DISCUSSION THREAD

**Discussion by mecrider** [ltsp/ltsp#676] [Q&A] [answered]
Is there a way to select which kernel gets used at boot-up by a thick client? Background: we have Ubuntu 20.04 on our server, chrootless. In December I installed tuxinvader's lts-mainline PPA and linux-generic-5.15 to get driver support for the sound module in a new Intel NUC. Yesterday I learned sound had quit again on that workstation. Turns out late last month the default 5.11 kernel got updated to 5.13, and that was now booting on thick clients. The server was still picking 5.15. I fixed it for now by updating to 5.16 from lts-mainline, but if the thick client is always picking the latest installed kernel that may not work after the next default kernel upgrade if we are still running the latest from lts-mainline.

**Comment by alkisg** [comment, marked as answer]
```console
$ ls -lha /boot/vmlinuz*
lrwxrwxrwx 1 root root  25 Φεβ   3 10:59 /boot/vmlinuz -> vmlinuz-5.15.0-18-generic
-rw------- 1 root root 11M Ιαν  13 18:14 /boot/vmlinuz-5.15.0-17-generic
-rw------- 1 root root 11M Ιαν  21 16:21 /boot/vmlinuz-5.15.0-18-generic
lrwxrwxrwx 1 root root  25 Φεβ   3 10:59 /boot/vmlinuz.old -> vmlinuz-5.15.0-17-generic
```

LTSP will prefer the /boot/vmlinuz symlink first if it exists, or the newer kernel if it doesn't exist.
If you install lower kernels afterwards, /boot/vmlinuz will point to them. In that case, update it manually before running `ltsp image /`.
